docs/spec/lsm/adr-002.md

```diff
@@ -0,0 +1,362 @@
+
+
+
+```proto
+// Params defines the parameters for the staking module.
+message Params {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  // unbonding_time is the time duration of unbonding.
+  google.protobuf.Duration unbonding_time = 1 [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+  // max_validators is the maximum number of validators.
+  uint32 max_validators = 2;
+  // max_entries is the max entries for either unbonding delegation or redelegation (per pair/trio).
+  uint32 max_entries = 3;
+  // historical_entries is the number of historical entries to persist.
+  uint32 historical_entries = 4;
+  // bond_denom defines the bondable coin denomination.
+  string bond_denom = 5;
+  // min_commission_rate is the chain-wide minimum commission rate that a validator can charge their delegators
+  string min_commission_rate = 6 [
+    (gogoproto.moretags)   = "yaml:\"min_commission_rate\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+  // validator_bond_factor is required as a safety check for tokenizing shares and 
+  // delegations from liquid staking providers
+  string validator_bond_factor = 7 [
+    (gogoproto.moretags) = "yaml:\"validator_bond_factor\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  // global_liquid_staking_cap represents a cap on the portion of stake that 
+  // comes from liquid staking providers
+  string global_liquid_staking_cap = 8 [
+    (gogoproto.moretags)   = "yaml:\"global_liquid_staking_cap\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+  // validator_liquid_staking_cap represents a cap on the portion of stake that 
+  // comes from liquid staking providers for a specific validator
+  string validator_liquid_staking_cap = 9 [
+    (gogoproto.moretags)   = "yaml:\"validator_liquid_staking_cap\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+}
+```
+
+### Data structures
+
+#### Validator
+The `Validator` struct had attributes `TotalExemptShares` and `TotalTokenizedShares` renamed to `TotalValidatorBondShares` and `TotalLiquidShares` respectively.
+
+```proto
+// Validator defines a validator, together with the total amount of the
+// Validator's bond shares and their exchange rate to coins. Slashing results in
+// a decrease in the exchange rate, allowing correct calculation of future
+// undelegations without iterating over delegators. When coins are delegated to
+// this validator, the validator is credited with a delegation whose number of
+// bond shares is based on the amount of coins delegated divided by the current
+// exchange rate. Voting power can be calculated as total bonded shares
+// multiplied by exchange rate.
+message Validator {
```

Comment:
ditto: this is outdated as it doesn't contain the changes for ICS. Maybe it's better to just leave the newly added fields and ignore the rest.

Reply:
In `adr.md` I've removed old fields, only showing newly added fields.